tests/periph/fmc/main.c

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (C) 2023 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @file
+ * @brief       Test for memories connected to the STM32 FMC/FSMC peripheral
+ *
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include "benchmark.h"
+#include "od.h"
+#include "periph_conf.h"
+
+#ifndef FMC_BANK
+#define FMC_BANK    0
+#endif
+
+#ifndef MEM_BUFSIZ
+#define MEM_BUFSIZ  1024
+#endif
+
+uint8_t mem_buf[MEM_BUFSIZ] = {};
+
+void bench_mem_write(uint32_t addr, uint32_t len)
+{
+    uint8_t *mem = (uint8_t *)addr;
+    uint32_t i = 0;
+
+    while (i < len) {
+        memcpy(mem + i, mem_buf, MEM_BUFSIZ);
+        i += MEM_BUFSIZ;
+    }
+}
+
+int main(void)
+{
+    printf("HCLK freq %lu MHz\n", CLOCK_AHB/MHZ(1));
+
+    uint8_t *data8  = (uint8_t *)(fmc_bank_config[FMC_BANK].address);
+    uint16_t *data16 = (uint16_t *)((fmc_bank_config[FMC_BANK].address) + 256);
+    uint32_t *data32 = (uint32_t *)((fmc_bank_config[FMC_BANK].address) +
+                                    (fmc_bank_config[FMC_BANK].size) - 256);
+
+    printf("8-bit data @%p, 16-bit data @%p, 32-bit data @%p\n",
+           data8, data16, data32);
+    puts("------------------------------------------------------------------------");
+
+    for (unsigned i = 0; i < 256; i++)
+    {
+        data8[i] = i;
+    }
+    od_hex_dump_ext(data8, 256, 16, fmc_bank_config[FMC_BANK].address);
+
+    for (unsigned i = 0; i < 256; i++)
+    {
+        if (data8[i] != i) {
+            printf("memory content did not match @%p\n", &data8[i]);
+            return 1;
+        }
+    }
```

Comment:
This file has to be uncrustified:

```suggestion
    for (unsigned i = 0; i < 256; i++) {
        data8[i] = i;
    }
    od_hex_dump_ext(data8, 256, 16, fmc_bank_config[FMC_BANK].address);

    for (unsigned i = 0; i < 256; i++) {
        if (data8[i] != i) {
            printf("memory content did not match @%p\n", &data8[i]);
            return 1;
        }
    }
```